model/protecode.py

```diff
@@ -35,4 +41,34 @@ def tls_verify(self):
 
 
 class ProtecodeCredentials(BasicCredentials):
-    pass
+    def auth_scheme(self):
+        return ProtecodeAuthScheme(self.raw.get('auth_scheme', 'basic_auth'))
+
+    def token(self):
+        return self.raw.get('token')
+
+    def validate(self):
+        if (auth_scheme := self.auth_scheme()) is ProtecodeAuthScheme.BASIC_AUTH:
+            if not (self.passwd() and self.username()):
+                raise ModelValidationError(
+                    'If basic auth is configured, both username and password are required'
+                )
+        elif auth_scheme is ProtecodeAuthScheme.OAUTH:
+            if not self.token():
+                raise ModelValidationError(
+                    'If oAuth is configured, a token is required'
```

Comment:
as indicated above. please do not mix those two credential-types into one class. (define one class per credential
consider defining the auth-type credential-class in the base.py-module, if this makes sense, but it is also OK for me to define it here.)